src/System.Linq.Parallel/tests/Combinatorial/CancellationParallelQueryCombinationTests.cs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using Xunit;
+
+namespace Test
+{
+    [Trait("category", "outerloop")]
+    public partial class ParallelQueryCombinationTests
+    {
+        [Theory]
+        [MemberData("UnaryOperators")]
+        [MemberData("BinaryOperators")]
+        public static void Aggregate_OperationCancelledException(LabelledOperation source, LabelledOperation operation)
+        {
+            CancellationTokenSource cs = new CancellationTokenSource();
```

Comment:
That was part of my follow-up point.  ```new CancellationToken(true).Equals(new CancellationToken(true))```, which is desirable in general but could lead to false positives if the library did the wrong thing.  By using a new CancellationTokenSource as you were doing originally, the only way the comparison on teh exception's Token would match is if it came from the same CTS, whereas with ```new CancellationToken(true)``` as the input, the OCE could manufacture another ```new CancellationToken(true)``` and it would match.